backend/app/core/cache.py

```
@@ -0,0 +1,161 @@
+
+    def __init__(self, max_size: int = 1000):
+        self._cache: dict[str, CacheEntry] = {}
+        self._lock = Lock()
+        self.max_size = max_size
+        self.hits = 0
+        self.misses = 0
+
+    def _generate_key(self, prefix: str, **kwargs) -> str:
+        """Generate a cache key from prefix and kwargs."""
+        key_str = f"{prefix}:{json.dumps(kwargs, sort_keys=True)}"
+        return hashlib.md5(key_str.encode()).hexdigest()
```

Ruff (0.14.13)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)